src/app/auth/profiles/profiles.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { AppConfig } from '../../../environments/environment';
+import { MatDialog } from '@angular/material';
+import {
+  JsonParseErrorDialogComponent
+} from './json-parse-error-dialog/json-parse-error-dialog.component';
+import ProfilesJson from './profiles.json';
+
+/**
+ * Indicates all the elements that make up a Profile.
+ */
+export interface Profile {
+  profileName: string;
+  username: string;
+  password: string;
+  encodedText: string;
+}
+
+/**
+ * This component is responsible allowing the user to select a Profile
+ * in the authentication page.
+ */
+
+@Component({
+  selector: 'app-profiles',
+  templateUrl: './profiles.component.html',
+  styleUrls: ['./profiles.component.css']
+})
+export class ProfilesComponent implements OnInit {
+
+  profiles: Profile[] = undefined; // List of profiles taken from profiles.json
+  blankProfile: Profile = {profileName: '', password: '', username: '', encodedText: ''}; // A blank profile to reset values
+
+  private readonly fs = require('fs');
+
+  // path of profile.json (same folder as app for production)
+  private filePath: string = __dirname.split('CATcher', 1)[0].concat('profiles.json');
+
+  @Output() selectedProfile: EventEmitter<Profile> = new EventEmitter<Profile>();
+
+  constructor(public errorDialog: MatDialog) { }
+
+  ngOnInit() {
+
+    // Developer Mode indicates that the profiles.json is internal.
+    if (this.isDeveloperMode()) {
+
+      this.profiles = ProfilesJson.profiles;
+
+      setTimeout(() => {
+        if (!this.isValid(this.profiles)) {
+          this.openErrorDialog();
+          this.profiles = undefined;
+        }
+      });
```

Comment:
do we really need this extra branch for developer mode? I prefer to minimize differentiating behavior between production mode and developer mode.

Reply:
Would this method be better?
```typescript
try {
      // If profile.json exists in app directory, read from it
      this.profiles = this.userProfileFileExists(this.filePath) ? JSON.parse(this.fs.readFileSync(this.filePath))['profiles']
        // Otherwise read from the profiles.json located in src/app/auth/profiles
        : ProfilesJson.profiles;
    } catch (parseError) {
      // Possible parsing error should the user configure their profiles.json incorrectly.
      console.log(parseError);
    }
```
Removing the need to check for if in `Developer` mode and nesting.